roles/dns-server-detect/tasks/main.yml

```diff
@@ -28,8 +29,9 @@
     public_dns_server: "{{ external_nsupdate_keys['public']['server'] }}"
   when:
     - external_nsupdate_keys['public'] is defined
+    - hostvars[groups['dns'][0]] is undefined
```

Comment:
I believe this will cause breakage when we want to use a (internally) provisioned DNS server for internal records + an external for external records.